pyanaconda/modules/common/structures/payload.py

```diff
@@ -0,0 +1,205 @@
+        return self._type
+
+    @type.setter
+    def type(self, url_type: Str):
+        """Set type of the url.
+
+        For list of possible values see type getter.
+        """
+        self._type = url_type
+
+
+class SSLConfigurationData(DBusData):
+    """Structure with SSL configuration settings."""
+
+    def __init__(self):
+        self._ca_cert_path = ""
+        self._client_cert_path = ""
+        self._client_key_path = ""
+
+    @property
+    def ca_cert_path(self) -> Str:
+        """Get CA certificate path."""
+        return self._ca_cert_path
+
+    @ca_cert_path.setter
+    def ca_cert_path(self, ca_cert_path: Str):
+        """Set CA certificate path."""
+        self._ca_cert_path = ca_cert_path
+
+    @property
+    def client_cert_path(self) -> Str:
+        """Get client certificate path."""
+        return self._client_cert_path
+
+    @client_cert_path.setter
+    def client_cert_path(self, client_cert_path: Str):
+        """Set client certificate path."""
+        self._client_cert_path = client_cert_path
+
+    @property
+    def client_key_path(self) -> Str:
+        """Get client key path."""
+        return self._client_key_path
+
+    @client_key_path.setter
+    def client_key_path(self, client_key_path: Str):
+        """Set client key path."""
+        self._client_key_path = client_key_path
+
+
+class RepoCostData(DBusData):
+    """Structure to store cost value of the repository with flag if this should be used."""
+
+    def __init__(self):
+        self._cost = 0
+        self._enabled = False
+
+    def set_cost(self, cost):
+        """Set cost value of this repository and set it enabled.
+
+        This is a shortcut to avoid calling enable = True all the time.
+
+        :param cost: cost value of this repository
+        :type cost: int
+        """
+        self.cost = cost
+        self.enabled = True
+
+    @property
+    def cost(self) -> Int:
+        """Get cost of this repository.
+
+        The relative cost of accessing this repository. This value is compared when the
+        priorities of two repositories are the same. The repository with the lowest cost is picked.
+        """
+        return self._cost
+
+    @cost.setter
+    def cost(self, cost: Int):
+        """Set cost of this repository.
+
+        The relative cost of accessing this repository. This value is compared when the
+        priorities of two repositories are the same. The repository with the lowest cost is picked.
+        """
+        self._cost = cost
+
+    @property
+    def enabled(self) -> Bool:
+        """Should the cost value be used.
+
+        If True then change the default repository value.
+        """
+        return self._enabled
+
+    @enabled.setter
+    def enabled(self, enabled: Bool):
+        """Set if the cost value should be used.
+
+        If True then change the default repository value.
+        """
+        self._enabled = enabled
+
+
+class PackagingConstrainsData(DBusData):
```

Comment:
I wonder whether we shouldn't have just one structure called `RepoConfiguration`. I know it looks pretty to have all these  classes, but there is not benefit for having them. It is much safer and easier to just set up one data object and call one DBus method. I would merge at least `URLData`, `PackagingConstrainsData`, `RepoCostData` and `SSLVerificationEnabled`.

Reply:
Fixed.